modules/build/src/main/scala/scala/build/preprocessing/directives/RequirePlatformsDirectiveHandler.scala

```diff
@@ -29,4 +33,53 @@ case object RequirePlatformsDirectiveHandler extends RequireDirectiveHandler {
       case None =>
         None
     }
+
+  override def handleValues(
+    directive: StrictDirective,
+    path: Either[String, Path],
+    cwd: ScopePath
+  ): Either[BuildException, (Option[BuildRequirements], Seq[Scoped[BuildRequirements]])] = {
+    val values          = DirectiveUtil.stringValues(directive.values, path, cwd)
+    val nonscopedValues = values.filter(_._3.isEmpty)
+    val scopedValues    = values.filter(_._3.nonEmpty)
+    val nonScopedPlatforms = Option(nonscopedValues.map(v => Platform.normalize(v._1)))
+      .filter(_.nonEmpty)
+
+    val nonscoped =
+      nonScopedPlatforms.fold[Either[BuildException, Option[BuildRequirements]]](Right(None)) {
```

Comment:
A `match` would be clearer than a `fold` here, IMO (no need to write an explicit type with a `match` I think).